sdk/core/azure-core-test/src/main/java/com/azure/core/test/TestBase.java

```diff
@@ -131,18 +146,55 @@ protected void beforeTest() {
     protected void afterTest() {
     }
 
+    /**
+     * Get test arguments need to run for the test framework based on the service version.
+     *
+     * @param serviceVersionList The service version argument for the parameterized tests.
+     * @param rollingServiceVersion Indicates whether the service version parameters need to rolling.
+     * @return Stream of arguments for parameterized test framework.
+     */
+    public static Stream<Arguments> getArgumentsFromServiceVersion(List<ServiceVersion> serviceVersionList,
+        boolean rollingServiceVersion) {
+        List<Arguments> argumentsList = new ArrayList<>();
+        int serviceVersionCount = serviceVersionList.size();
+        List<HttpClient> httpClientList = getHttpClients();
+        int httpClientCount = httpClientList.size();
+        boolean rollingHttpClient = HTTP_CLIENT_FROM_ENV.equalsIgnoreCase(AZURE_TEST_HTTP_CLIENTS_VALUE_ROLLING);
+        for (ServiceVersion s: serviceVersionList) {
+            for (HttpClient h: httpClientList) {
+                argumentsList.add(Arguments.of(h, s));
+            }
+        }
```

Comment:
You need this n x m matrix only if both are http clients and service versions are rolling. If only one of them is rolling, then creating this matrix is unnecessary.

Reply:
I am trying to make the code structural, so I made the matrix first and then filter out by needs.

As this is deterministic matrix, it does not increase time complexity too much but saves tons of codes.